sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -105,26 +105,60 @@ object GpuParquetFileFormat {
     }
   }
 
-  def parquetWriterOptionsFromSchema[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions]
-  (builder: ParquetColumnWriterOptions.NestedBuilder[T, V],
-   schema: StructType, writeInt96: Boolean): T = {
+  def getParquetColumnWriterOption(
```

Comment:
Why exactly do we need a separate API here just for map keys and values?  I really don't like the existing code for arrays that pulls out the element type to wrap it in a struct just so we can have a little bit of code reuse. To me it feels very much like we are doing the recursion at the wrong level, but I have not taken the time to think it through.